custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -78,9 +82,32 @@ def _predict_or_transform(self, logger=None):
             return {"message": "ERROR: " + wrong_key_error_message}, response_status
 
         if self._target_type == TargetType.TRANSFORM:
-            out_data = self._predictor.transform(
-                binary_data=binary_data, mimetype=mimetype, charset=charset
-            )
+
+            if target_filestorage is not None:
+                target_binary_data = target_filestorage.stream.read()
+                target_mimetype = StructuredInputReadUtils.resolve_mimetype_by_filename(
+                    target_filestorage.filename
+                )
+
+                if logger is not None:
+                    logger.debug(
+                        "Target filename provided under y key: {}".format(
+                            target_filestorage.filename
+                        )
+                    )
+
+                out_data, out_target = self._predictor.transform(
+                    binary_data=binary_data,
```

Comment:
Just as a note.... Looks like we need some structured DTO to assemble and pass data within different flows

Reply:
agreed

Reply:
added some documentation, but there might be more to add in some spots.... @4dahalibut also may want to weigh in on doc updates before this goes in

Reply:
yeah I mean, I wouldn't mind a refactor to get rid of some of these if statements tbh

Reply:
I think maybe it makes sense to refactor `_predict_or_transform` into separate methods, one for predict and one for transform